src/arc_utilities/transformation_helper.py

```diff
@@ -404,4 +412,23 @@ def TransformFromDH(d, a, theta, alpha):
 
 def PoseFromDH(d, a, theta, alpha):
     tfmatrix = BuildMatrixFromDH(d, a, theta, alpha)
-    return PoseFromMatrix(tfmatrix)
\ No newline at end of file
+    return PoseFromMatrix(tfmatrix)
+
+
+def spherical_to_vector3(r, phi, theta):
+    # https://en.wikipedia.org/wiki/Spherical_coordinate_system#Cartesian_coordinates
+    x = r * np.sin(theta) * np.cos(phi)
+    y = r * np.sin(theta) * np.sin(phi)
+    z = r * np.cos(theta)
+    return [x, y, z]
+
+
+def vector3_to_spherical(v):
```

Comment:
Seems kind of weird that the inverse function expects separate scalar inputs but this one returns tuple output and takes vector input. Not sure what the best option would be; having separate scalar arguments is nice for IDE hints; maybe this one also takes separate scalars as inputs (but then having to specify x, y, z would become annoying...)? I'd be OK if you want to keep it this way, just wanted to point out that there's some asymmetry in the API.

Reply:
Good point! The user will almost always have [x,y,z] as a list or numpy array, so I think we should keep this signature. Therefore I think it would make sense to change the other function to take a list as well, and put the order in a docstring.